emission/analysis/modelling/tour_model/featurization.py

```diff
@@ -1,150 +1,137 @@
-    #the same by the clustering algorithm
-    def map_individuals(self):
-        import pygmaps
-        from matplotlib import colors as matcol
-        colormap = plt.cm.get_cmap()
-        import random 
-        r = random.sample(range(len(set(self.labels))), len(set(self.labels)))
-        rand = []
-        for i in range(len(self.labels)):
-            rand.append(r[self.labels[i]]/float(self.clusters))
-        for color in set(self.colors):
-            first = True
-            num_paths = 0
-            for i in range(len(self.colors)):
-                if self.colors[i] == color:
-                    num_paths += 1
-                    start_lat = self.data[i]['trip_start_location'][1]
-                    start_lon = self.data[i]['trip_start_location'][0]
-                    end_lat = self.data[i]['trip_end_location'][1]
-                    end_lon = self.data[i]['trip_end_location'][0]
-                    if first:
-                        mymap = pygmaps.maps(start_lat, start_lon, 10)
-                        first = False
-                    path = [(start_lat, start_lon), (end_lat, end_lon)]
-                    mymap.addpath(path, matcol.rgb2hex(colormap(rand[i])))
-            if num_paths > 1:
-                mymap.draw('./mycluster' + str(color) + '.html')
-            else:
-                mymap.draw('./onemycluster' + str(color) + '.html') #clusters with only one trip
-
-    #plot all the clusters on the map. Outputs mymap.html, a map with colors defined by the ground 
-    #truth, and mylabels.html, a map with colors defined by the clustering algorithm. 
-    def map_clusters(self):
-        import pygmaps
-        from matplotlib import colors as matcol
-        colormap = plt.cm.get_cmap()
-        mymap = pygmaps.maps(37.5, -122.32, 10)
-        for i in range(len(self.points)):
-            start_lat = self.data[i]['trip_start_location'][1]
-            start_lon = self.data[i]['trip_start_location'][0]
-            end_lat = self.data[i]['trip_end_location'][1]
-            end_lon = self.data[i]['trip_end_location'][0]
-            path = [(start_lat, start_lon), (end_lat, end_lon)]
-            mymap.addpath(path, matcol.rgb2hex(colormap(float(self.colors[i])/len(set(self.colors)))))
-        mymap.draw('./mymap.html')
-        mymap2 = pygmaps.maps(37.5, -122.32, 10)
-        for i in range(len(self.points)):
-            start_lat = self.data[i]['trip_start_location'][1]
-            start_lon = self.data[i]['trip_start_location'][0]
-            end_lat = self.data[i]['trip_end_location'][1]
-            end_lon = self.data[i]['trip_end_location'][0]
-            path = [(start_lat, start_lon), (end_lat, end_lon)]
-            mymap2.addpath(path, matcol.rgb2hex(colormap(float(self.labels[i])/self.clusters)))
-        mymap2.draw('./mylabels.html')
-
-
+# Standard imports
+import matplotlib
+matplotlib.use('Agg')

```

Comment:
Nevermind, I'm using it in map_clusters to get the color map. I can move the import to inside the function if that would be better.